homeassistant/components/recorder/util.py

```diff
@@ -169,14 +169,24 @@ def validate_sqlite_database(dbpath: str) -> bool:
     return True
 
 
-def run_checks_on_open_db(dbpath, cursor):
+def run_checks_on_open_db(dbpath, cursor, db_integrity_check):
     """Run checks that will generate a sqlite3 exception if there is corruption."""
     if basic_sanity_check(cursor) and last_run_was_recently_clean(cursor):
         _LOGGER.debug(
             "The quick_check will be skipped as the system was restarted cleanly and passed the basic sanity check"
         )
         return
 
+    if not db_integrity_check:
+        # Always warn so when it does fail they remember it has
+        # been manually disabled
+        _LOGGER.warning(
```

Comment:
We always warn when the check is skipped to ensure
that the user remembers if a long time has passed between
when they disabled the check and when the database fails
since the MTBF can be quite high.